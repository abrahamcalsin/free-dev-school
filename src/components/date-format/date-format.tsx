import "dayjs/locale/en";
import "dayjs/locale/es";

import * as React from "react";
import { Text } from "@chakra-ui/react";
import dayjs from "dayjs";

import { dateFormat } from "~/utils";

export interface DateFormatProps {
  date: Parameters<typeof dayjs>["0"];
  locale: "en" | "es";
}

export function DateFormat(props: DateFormatProps) {
  const { date, locale } = props;

  const formattedDate = dayjs(date).locale(locale).format(dateFormat.USA);

  return (
    <Text
      as="span"
      display="inline-block"
      fontSize={{ base: "x-small", sm: "xs" }}
      fontWeight="semibold"
      textColor="gray.400"
    >
      {formattedDate}
    </Text>
  );
}

DateFormat.defaultProps = {
  locale: "en",
};
