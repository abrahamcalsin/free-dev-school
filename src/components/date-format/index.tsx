import * as React from "react";
import { Text } from "@chakra-ui/react";
import dayjs from "dayjs";
import en from "dayjs/locale/en";

import { dateFormat } from "~/utils";

interface DateFormatProps {
  videoDate?: string;
}

export default function DateFormat(props: DateFormatProps) {
  const { videoDate = new Date() } = props;

  const date = dayjs(videoDate).locale(en).format(dateFormat.video);

  return (
    <Text
      as="span"
      display="inline-block"
      fontSize={{ base: "x-small", sm: "xs" }}
      fontWeight="semibold"
      textColor="gray.400"
    >
      {date}
    </Text>
  );
}
