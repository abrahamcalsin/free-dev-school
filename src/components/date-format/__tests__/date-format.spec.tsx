import { render, screen } from "@testing-library/react";

import { DateFormat, DateFormatProps } from "../date-format";

const setup = (props: DateFormatProps) => render(<DateFormat {...props} />);

describe("DateFormat", () => {
  it("shows the formatted date in spanish", () => {
    const view = setup({
      date: "Sep 24 2022",
      locale: "es",
    });

    const text = screen.queryByText(/septiembre 24, 2022/i);

    expect(text).toBeInTheDocument();

    expect(view.baseElement).toMatchSnapshot();
  });

  it("shows the formatted date in english", () => {
    const view = setup({
      date: "Sep 25 2022",
      locale: "en",
    });

    const text = screen.queryByText(/september 25, 2022/i);

    expect(text).toBeInTheDocument();

    expect(view.baseElement).toMatchSnapshot();
  });
});
