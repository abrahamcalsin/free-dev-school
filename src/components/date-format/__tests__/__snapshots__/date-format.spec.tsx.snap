// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFormat shows the formatted date in english 1`] = `
<body>
  <div>
    <span
      class="chakra-text css-0"
    >
      September 25, 2022
    </span>
  </div>
</body>
`;

exports[`DateFormat shows the formatted date in spanish 1`] = `
<body>
  <div>
    <span
      class="chakra-text css-0"
    >
      septiembre 24, 2022
    </span>
  </div>
</body>
`;
